apps/web/src/pages/integrations/IntegrationsStoreModal.tsx

```diff
@@ -0,0 +1,339 @@
+import { useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import {
+  ChannelTypeEnum,
+  IConfigCredentials,
+  ILogoFileName,
+  providers,
+  PushProviderIdEnum,
+  EmailProviderIdEnum,
+  InAppProviderIdEnum,
+} from '@novu/shared';
+import { Grid, Group, Modal, ActionIcon, useMantineColorScheme, createStyles, MantineTheme } from '@mantine/core';
+import * as cloneDeep from 'lodash.clonedeep';
+import { useIntegrations } from '../../hooks';
+import { When } from '../../components/utils/When';
+import { NovuEmailProviderModal } from './components/NovuEmailProviderModal';
+import { NovuInAppProviderModal } from './components/NovuInAppProviderModal';
+import { ChannelGroup } from './components/Modal/ChannelGroup';
+import { colors, shadows, Title } from '../../design-system';
+import { ConnectIntegrationForm } from './components/Modal/ConnectIntegrationForm';
+import { Close } from '../../design-system/icons/actions/Close';
+
+export function IntegrationsStoreModal({
+  openIntegration,
+  closeIntegration,
+}: {
+  openIntegration: boolean;
+  closeIntegration: () => void;
+}) {
+  const { integrations, loading: isLoading, refetch } = useIntegrations();
+  const { colorScheme } = useMantineColorScheme();
+  const [emailProviders, setEmailProviders] = useState<IIntegratedProvider[]>([]);
+  const [smsProvider, setSmsProvider] = useState<IIntegratedProvider[]>([]);
+  const [chatProvider, setChatProvider] = useState<IIntegratedProvider[]>([]);
+  const [pushProvider, setPushProvider] = useState<IIntegratedProvider[]>([]);
+  const [isFormOpened, setFormIsOpened] = useState(false);
+  const [isCreateIntegrationModal, setIsCreateIntegrationModal] = useState(false);
+  const [provider, setProvider] = useState<IIntegratedProvider | null>(null);
+  const [selectedProvider, setSelectedProvider] = useState<string>('');
+
+  const { classes } = useModalStyles();
+
+  async function handlerVisible(
+    visible: boolean,
+    createIntegrationModal: boolean,
+    providerConfig: IIntegratedProvider
+  ) {
+    setFormIsOpened(visible);
+    setProvider(providerConfig);
+    setSelectedProvider(providerConfig.providerId);
```

Comment:
if you set the provider in the state then you don't need a separate state just for the providerId, as you can take it from the provider object